modules/openapi-generator/src/main/resources/go-server/controller-api.mustache

```diff
@@ -612,9 +612,9 @@ func (c *{{classname}}Controller) {{nickname}}(w http.ResponseWriter, r *http.Re
 	{{#isBodyParam}}
 	var {{paramName}}Param {{dataType}}
 	d := json.NewDecoder(r.Body)
-	{{^isAdditionalPropertiesTrue}}
+	{{#strictResponseDecoding}}
 	d.DisallowUnknownFields()
-	{{/isAdditionalPropertiesTrue}}
+	{{/strictResponseDecoding}}
 	if err := d.Decode(&{{paramName}}Param); err != nil {{^required}}&& !errors.Is(err, io.EOF) {{/required}}{
```

Comment:
thanks for the pr

before introducing another option, did you try setting `disallowAdditionalPropertiesIfNotPresent` to false (or true which should be the default)?